aten/src/ATen/native/mkldnn/xpu/detail/Matmul.cpp

```diff
@@ -255,4 +255,139 @@ sycl::event matmul(
   return matmul_event;
 }
 
+sycl::event scaled_matmul(
+    at::Tensor& result,
+    const at::Tensor& mat1,
+    const at::Tensor& mat2,
+    const c10::optional<at::Tensor>& bias,
+    const at::Tensor& scale_a,
+    const at::Tensor& scale_b,
+    Attr attr,
+    const std::vector<sycl::event>& deps = {}) {
+  at::Device cur_device = at::Device(at::kXPU, c10::xpu::current_device());
+  dnnl::engine engine = GpuEngineManager::Instance().get_engine(cur_device);
+  auto stream = GpuStreamManager::Instance().get_stream();
```

Comment:
```suggestion
  auto& engine = GpuEngineManager::Instance().get_engine();
  auto& stream = GpuStreamManager::Instance().get_stream();
```
Based on https://github.com/pytorch/pytorch/pull/147349, we could make the code easier.